exporter/v1_compatibility.go

```diff
@@ -1200,9 +1222,118 @@ func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric,
 }
 
+func balancerRunning(ctx context.Context, client *mongo.Client) (prometheus.Metric, error) {
+	type bss struct {
+		InBalancerRound bool `bson:"inBalancerRound"`
+	}
+	var bs bss
+	running := 0
+	err := client.Database("admin").RunCommand(ctx, bson.M{"balancerStatus": 1}).Decode(&bs)
+	if err != nil {
+		return nil, err
+	}
+
+	if bs.InBalancerRound {
+		running = 1
+	}
+	name := "mongodb_mongos_sharding_balancer_running"
+	help := "Balancer is running"
+
+	d := prometheus.NewDesc(name, help, nil, nil)
+	metric, err := prometheus.NewConstMetric(d, prometheus.GaugeValue, float64(running))
+
+	return metric, err
+}
+
+func migratingCollections(ctx context.Context, client *mongo.Client) ([]prometheus.Metric, error) {
+	cursor, err := client.Database("config").Collection("migrations").Find(ctx, bson.M{})
+	if err != nil {
+		return nil, err
+	}
+
+	var migrations []bson.M
+	if err = cursor.All(ctx, &migrations); err != nil {
+		return nil, err
+	}
+	metrics := make([]prometheus.Metric, 0, len(migrations))
+
+	var migratingCollections []string
+
+	type bss struct {
+		When time.Time `bson:"when"`
+	}
+	var bs bss
+	for _, migration := range migrations {
+		name := "mongodb_mongos_sharding_migrating"
+		help := "Collection has active migration"
+
+		client.Database("config").Collection("locks").FindOne(ctx, bson.M{"_id": migration["ns"].(string),
+			"why": "Migrating chunk(s) in collection testdb.testcol"}).Decode(&bs)
+		labels := map[string]string{
+			"ns":        migration["ns"].(string),
+			"fromShard": migration["fromShard"].(string),
+			"toShard":   migration["toShard"].(string),
+			"startedAt": bs.When.String(),
+		}
+		d := prometheus.NewDesc(name, help, nil, labels)
+		metric, err := prometheus.NewConstMetric(d, prometheus.GaugeValue, float64(1))
+		if err != nil {
+			continue
+		}
+
+		metrics = append(metrics, metric)
+		migratingCollections = append(migratingCollections, migration["ns"].(string))
+	}
+	// shardedCollections, err := getShardedCollections(ctx, client)
+	// if err != nil {
+	// 	return metrics, err
+	// }
+
+	// nonMigratingCollections := difference(shardedCollections, migratingCollections)
+	// for _, nonMigratingCollection := range nonMigratingCollections {
+	// 	name := "mongodb_mongos_sharding_migrating"
+	// 	help := "Collection has active migration"
+	// 	labels := map[string]string{
+	// 		"ns": nonMigratingCollection,
+	// 	}
+	// 	d := prometheus.NewDesc(name, help, nil, labels)
+	// 	metric, err := prometheus.NewConstMetric(d, prometheus.GaugeValue, float64(0))
+	// 	if err != nil {
+	// 		continue
+	// 	}
+
+	// 	metrics = append(metrics, metric)
+	// }
+	return metrics, nil
+}
+
+func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric, error) {
+	n, err := client.Database("config").Collection("chunks").CountDocuments(ctx, bson.M{})
+	if err != nil {
+@ -1200,9 +1317,26 @@ func chunksTotal(ctx context.Context, client *mongo.Client) (prometheus.Metric,
```

Comment:
this line looks pretty weird

Reply:
I tried running make format but got these issues:

go mod tidy
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
make: *** [format] Error 1

Reply:
What version of golang do you have locally?

Reply:
1.16.5

Reply:
Please update to at least 1.17.